fidesctl/src/fidesapi/crud.py

```diff
@@ -6,30 +6,61 @@
 
 from typing import List, Dict
 
-from fastapi import APIRouter, status
+from fastapi import APIRouter, HTTPException, status
 from sqlalchemy import update as _update
 from sqlalchemy.dialects.postgresql import Insert as _insert
+from sqlalchemy.exc import SQLAlchemyError
 
 from fidesapi import db_session
 from fidesapi.sql_models import sql_model_map, SqlAlchemyBase
 from fideslang import model_map
 
 
+class NotFoundError(HTTPException):
+    """
+    To be raised when a requested resource does not exist in the database.
+    """
+
+    def __init__(self, resource_type: str, fides_key: str) -> None:
+        detail = {
+            "error": "resource does not exist",
+            "resource_type": resource_type,
+            "fides_key": fides_key,
+        }
+
+        super().__init__(status.HTTP_404_NOT_FOUND, detail=detail)
+
+
+class QueryError(HTTPException):
+    """
+    To be raised when a database query fails.
+    """
+
+    def __init__(self, err: SQLAlchemyError) -> None:
+        detail = {"error": str(err.__dict__["orig"])}
```

Comment:
I don't like this at all. It looks ugly, and more importantly, it leaks implementation details. I struggled mightily to read into the error instance and create something nicer, but it was very challenging.

Reply:
i'll take a stab at this over the weekend and see if I can think of something. It never feels good to ship code you have reservations about so let's see if we can get this looking nicer or just stick with the initial fix you made for now and we can revisit this in #241